services/QuillLMS/app/workers/find_district_standards_reports_worker.rb

```diff
@@ -4,13 +4,18 @@ class FindDistrictStandardsReportsWorker
   include Sidekiq::Worker
   sidekiq_options queue: SidekiqQueue::CRITICAL
 
-  def perform(admin_id)
+  def perform(admin_id, is_fremium_view)
     return unless admin_id
 
     serialized_district_standards_reports_cache_life = 60*60*25
-    serialized_district_standards_reports = ProgressReports::DistrictStandardsReports.new(admin_id).results.to_json
-    $redis.set("#{SchoolsAdmins::DISTRICT_STANDARD_REPORTS_CACHE_KEY_STEM}#{admin_id}", serialized_district_standards_reports)
-    $redis.expire("#{SchoolsAdmins::DISTRICT_STANDARD_REPORTS_CACHE_KEY_STEM}#{admin_id}", serialized_district_standards_reports_cache_life)
-    PusherDistrictStandardsReportsCompleted.run(admin_id)
+    serialized_district_standards_reports = ProgressReports::DistrictStandardsReports.new(admin_id, is_fremium_view).results.to_json
+    if is_fremium_view
+      $redis.set("#{SchoolsAdmins::FREEMIUM_DISTRICT_STANDARD_REPORTS_CACHE_KEY_STEM}#{admin_id}", serialized_district_standards_reports)
```

Comment:
Similar to `del` you'll want to rewrite this with Rails wrapper [write](https://api.rubyonrails.org/v7.0.4/classes/ActiveSupport/Cache/Store.html#method-i-write).  Note the `set` and `expire` are bundled into one call

```
Rails.cache.write(
  "#{SchoolsAdmins::FREEMIUM_DISTRICT_STANDARD_REPORTS_CACHE_KEY_STEM}#{admin_id}", 
  serialized_district_standards_reports, 
  expires_in: serialized_district_standards_reports_cache_life
)
```